lib/ansible/modules/cloud/openstack/os_port.py

```diff
@@ -355,11 +354,10 @@ def main():
                 for v in module.params['security_groups']
             ]
 
-        if module.params['vnic_type']:
-            # Neutron API accept 'binding:vnic_type' as an argument
-            # for the port type.
-            module.params['binding:vnic_type'] = module.params['vnic_type']
-            module.params.pop('vnic_type', None)
+        # Neutron API accept 'binding:vnic_type' as an argument
+        # for the port type.
+        module.params['binding:vnic_type'] = module.params['vnic_type']
+        module.params.pop('vnic_type', None)
```

Comment:
Couple notes:  You probably shouldn't pop from module.params.... I usually consider that to be a "constant" value.  It's the record of what the user specified in the playbook for the module to look at but not to change.

If you do want to pop it, you don't need to do it as two steps.  This should be equivalent to what the present code does:
``` python
        module.params['binding:vnic_type'] = module.params.pop('vnic_type', None)
```

Reply:
I tried to keep the code change as minimal as possible since this just a bugfix.
Additionally, I am not sure, if it is a good idea to modify a backport. Maybe I can create a code change in the devel branch and add this to this PR?

Reply:
Yes, that would be the way to do it.  Also, just for clarity, those are not blockers for merging this PR.  The question of behaviour change is the only thing holding this up.

Reply:
@abadger Please find the requested change in https://github.com/ansible/ansible/pull/57838